tests/test_adaptor_strict.cc

```diff
@@ -40,88 +40,121 @@ namespace rascal {
+      int pair_counter{};
+      constexpr bool verbose{false};
 
-      auto type{atom.get_atom_type()};
-      BOOST_CHECK_EQUAL(type, this->fixture.atom_types[index]);
-      ++atom_counter;
+      for (auto atom : adaptor_strict) {
+        auto index{atom.get_global_index()};
+        BOOST_CHECK_EQUAL(index, atom_counter);
 
-      for (auto pair : atom) {
-        auto pair_offset{pair.get_global_index()};
-        auto pair_type{pair.get_atom_type()};
-        if (verbose) {
-          std::cout << "pair (" << atom.back() << ", " << pair.back()
-                    << "), pair_counter = " << pair_counter
-                    << ", pair_offset = " << pair_offset
-                    << ", atom types = " << type << ", " << pair_type
-                    << std::endl;
-        }
+        auto type{atom.get_atom_type()};
+        BOOST_CHECK_EQUAL(type, atom_types[index]);
+        ++atom_counter;
+
+        for (auto pair : atom.with_self_pair()) {
+          auto pair_offset{pair.get_global_index()};
+          auto pair_type{pair.get_atom_type()};
+          if (verbose) {
+            std::cout << "pair (" << atom.back() << ", " << pair.back()
+                      << "), pair_counter = " << pair_counter
+                      << ", pair_offset = " << pair_offset
+                      << ", atom types = " << type << ", " << pair_type
+                      << std::endl;
+          }
 
-        BOOST_CHECK_EQUAL(pair_counter, pair_offset);
-        ++pair_counter;
+          BOOST_CHECK_EQUAL(pair_counter, pair_offset);
+          ++pair_counter;
+        }
       }
     }
   }
+
   /* ---------------------------------------------------------------------- */
-  /*
-   * Compare the strict neighbour list with the linked cell one
-   * selecting only the atoms within a cutoff radius
-   *
-   * ``manager`` is a MaxOrder=1 StructureManager
+  /**
+   * Test that the atom index from a neighbour matches the atom tag of the
+   * ClusterRefKey returned by get_atom_j
    */
-  BOOST_FIXTURE_TEST_CASE(strict_test,
-                          ManagerFixture<StructureManagerCenters>) {
+  BOOST_FIXTURE_TEST_CASE_TEMPLATE(get_atom_j_test, Fix, multiple_fixtures,
+                                   Fix) {
+    auto && managers = Fix::managers;
+    constexpr bool verbose{false};
+    for (auto & manager : managers) {
+      double cutoff{manager->get_cutoff()};
+      auto adaptor_strict{make_adapted_manager<AdaptorStrict>(manager, cutoff)};
+      adaptor_strict->update();
+
+      for (auto atom : adaptor_strict) {
+        for (auto pair : atom) {
+          auto atom_j_index = adaptor_strict->get_atom_index(pair.back());
+          auto atom_j = pair.get_atom_j();
+          auto atom_j_tag = atom_j.get_atom_tag_list();
+          if (verbose) {
+            std::cout << "neigh: " << atom_j_index
+                      << " tag_j: " << atom_j_tag[0] << std::endl;
+          }
+
+          BOOST_CHECK_EQUAL(atom_j_index, atom_j_tag[0]);
```

Comment:
remark: be always aware that this test is only valid with SMC as root implementation; I added to comment.